app/controllers/api/v1/apis_controller.rb

```diff
@@ -0,0 +1,15 @@
+class Api::V1::ApisController < ApplicationController
+  before_filter :authorize_app_secret
+
+  private
+
+  def authorize_app_secret
+    unless correct_app_secret?
+      render nothing: true, status: 404
+    end
+  end
+
+  def correct_app_secret?
+    request.headers['X-App-Secret'] == ENV.fetch('HEADER_SECRET')
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.